packages/rich-text/src/create.js

```diff
@@ -36,9 +34,23 @@ function simpleFindKey( object, value ) {
 }
 
 function toFormat( { type, attributes } ) {
-	const formatType = find( getFormatTypes(), ( { match } ) =>
-		type === match.tagName
-	);
+	let formatType;
+
+	if ( attributes && attributes.class ) {
+		formatType = select( 'core/rich-text' ).getFormatTypeForClassName( attributes.class );
```

Comment:
Are these two places here the only places where we rely globally on the data module? 
I'd like us to avoid this pattern in the future. And only use the data module in withSelect/withDispatch or provide the registry as an argument somehow.

The issue with this pattern is that it relies on a singleton which is not great if you have the same npm module installed twice for instance.

I'm fine keeping them for now, if we already do the same elsewhere in `rich-text`.

Reply:
It got the data the same way before, jut through a wrapper function. Sounds good to revisit separately.

Reply:
Can't we just change the signature of the function to `toFormat({ type, attributes }, availableFormats)`? In my testing we do this in this function and in `fromFormat`. I'd prefer if we can avoid breaking changes here by adding this extra argument, but if you think it's a big refactoring, let's delay.

Reply:
Where should the available formats be given? in the RichText component? And then we pass it down all the way `toHTMLString( { availableFormats  } )` etc.?

Reply:
Well, yes I would prefer this or rewrite all these APIs as selectors. It does feel more impactful that I first thought so you have my 👍 for shipping as is.

Reply:
This would also make writing tests easier as you don't have to include stores in tests. This issue also exists in other places. I agree that we should keep such usage to the absolute minimum. We can iterate on the codebase later.

Reply:
Thinking a bit more about it, I think that it might be better to avoid introducing more specialized selectors `getFormatTypeForClassName` and `getFormatTypeForBareElement. Instead, we can perform filtering using utility helpers in this file. I don't think we will need this logic anywhere else.